packages/js/src/utils/errors.ts

```diff
@@ -0,0 +1,12 @@
+export enum LogtoErrorCode {
+  invalidIdTokenIat = 'invalid issued at time',
+}
+
+export class LogtoError extends Error {
+  code: string;
+
+  constructor(code: string, message?: string) {
+    super(message);
+    this.code = code;
+  }
+}
```

Comment:
1. `code` should be typed, again, please refer to `RequestError` in `logto/packages/core/src/errors/RequestError/index.ts`
2. instead of having an enum as type, we'd prefer a [literal type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) here since it has more flexibility
3. error code values can follow the convention defined in `RequestError`

Reply:
Fixed, PTAL.